### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by ruwanta**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/org.wso2.carbon.identity.conditional.auth.functions.test.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/test/utils/sequence/JsSequenceHandlerRunner.java [hunk 1]
```diff
@@ -0,0 +1,653 @@
...
+                                           "Test");
+            result.getUserAttributes().put(ClaimMapping
+                                                   .build("http://wso2.org/claims/lastname",
+                                                          "http://wso2.org/claims/lastname", "Test", false),
+                                           "User");
+            return result;
+        }
+    }
+
+    private static class MockServletRequest implements HttpServletRequest {
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.conditional.auth.functions.test.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/test/utils/sequence/JsSequenceHandlerRunner.java]: Wouldn't it be better to mock this using mockito as required?

File: components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml [hunk 2]
```diff
@@ -66,6 +67,25 @@
...
+        <dependency>
+            <groupId>com.h2database.wso2</groupId>
+            <artifactId>h2-database-engine</artifactId>
+            <version>${com.h2database.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon</groupId>
+            <artifactId>org.wso2.carbon.utils</artifactId>
+            <version>${carbon.kernel.version}</version>
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml]: Move the versions to the parent pom (under dependency mgt)

File: components/org.wso2.carbon.identity.conditional.auth.functions.test.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/test/utils/sequence/JsSequenceHandlerRunner.java (hunk 1 above)
Comment (ruwanta) [reply components/org.wso2.carbon.identity.conditional.auth.functions.test.utils/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/test/utils/sequence/JsSequenceHandlerRunner.java]: @pulasthi7 using mockito will cause dependency issues when the dependent project uses different version. Hence it was intentionally not done.

File: components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml (hunk 2 above)
Comment (ruwanta) [reply components/org.wso2.carbon.identity.conditional.auth.functions.user/pom.xml]: Done with 99fb4d3713f6ef978f44d526785443a3827b3442

**Comment by pulasthi7** [comment]
Conflicts resolved and fixed some other failures with https://github.com/wso2-extensions/identity-conditional-auth-functions/pull/26

**Pull request closed without merging**